romancal/tessel_match/tessel_match.py

```diff
@@ -0,0 +1,291 @@
+        equivalent 2-d numpy array, or
+        a GWCS instance. The instance must have either the bounding_box or
+        pixel_shape attribute defined, or the following image_shape argument
+        must be supplied
+
+    image_shape : image shape to be used if a GWCS instance is supplied
+        and does not contain a value for either the bounding_box or
+        pixel_shape attributes. Default value is None.
+
+    Returns
+    -------
+    A sequence of the indices of all tessels that overlap the supplied image
+    (in the referenced tessel table). The indices may be used to obtain all
+    necessary information about the tessels.
+    """
+
+    if isinstance(image_corners, wcs.WCS):
+        iwcs = image_corners
+        # Now must find size of correspinding image, with three possible
+        # sources of that information.
+        if (
+            (not hasattr(iwcs, "bounding_box") or iwcs.bounding_box is None)
+            and (not hasattr(iwcs, "pixel_shape") or iwcs.pixel_shape is None)
+            and image_shape is None
+        ):
+            raise ValueError(
+                "Use of a wcs object requires at least one of the bounding_box"
+                " or pixel_shape attributes be set to the image shape or that the"
+                "image_shape argument be set"
+            )
+        if image_shape is not None:
+            pass
+        else:
+            if hasattr(iwcs, "bounding_box") and iwcs.bounding_box is not None:
+                # Presumes that the bounding_box matches image array boundaries
+                bbintervals = iwcs.bounding_box.intervals
+                # This compensates for the half pixel adjustment in the general code.
+                image_shape = (bbintervals[0].upper + 0.5, bbintervals[1].upper + 0.5)
+            elif hasattr(iwcs, "pixel_shape") and iwcs.pixel_shape is not None:
+                image_shape = iwcs.pixel_shape
+        # Compute the image corners ra, dec from the wcs
+        (cxm, cxp), (cym, cyp) = (
+            (-0.5, image_shape[1] - 0.5),
+            (-0.5, image_shape[0] - 0.5),
+        )
+        image_corners = (iwcs(cxp, cyp), iwcs(cxm, cyp), iwcs(cxm, cym), iwcs(cxp, cym))
+    ttab = TESSEL_TABLE
+    ra = ttab[:]["ra_center"]
+    dec = ttab[:]["dec_center"]
+    # # Convert all celestial coordinates to cartesion coordinates.
+    vec_centers = np.array(sgv.lonlat_to_vector(ra, dec)).transpose()
+    # # Organize corners into two ra, dec lists
+    # imra = np.array([point[0] for point in image_corners])
+    # imdec = np.array([point[1] for point in image_corners])
+    vec_im_corners = image_coords_to_vec(image_corners)
+    # Approximate center of image by averaging corner vectors
+    im_center = normalize_vector(vec_im_corners.mean(axis=1))
+    # Compute difference vector between image centerand tessel centers
+    diff = vec_centers - im_center
+    dist = np.sqrt((diff**2).sum(axis=1)) * 180 / np.pi
+    match = np.where(dist < 0.5)
```

Comment:
Is 0.5 here a magic constant?  Would it make sense to pick something derived from the max linear size of the skycells or image?

Reply:
No, 0.5 isn't magic and it is certainly way larger than it needs to be. Picking an optimum value may need experimentation or as you describe. It doesn't cost a lot to be a little conservative on it.

Reply:
As for adding some buffer region around the given corners, I had the same thought. I wouldn't put that in the table, but have the code do that (something like the equivalent to parallel sides to the existing sides but bumped out some fixed amount. It's not trivial, but not a big deal to implement on the fly.

Reply:
Why not add extra vertices to the `SphericalPolygon` instead of padding?

Reply:
Then the question becomes how many do you need and that may still leave (for convex borders) edge (ahem) cases that happen rarely. I heard from someone that spherical_geometry speed decreases rapidly with many vertices. I'll try to remember who that was.

Reply:
It also makes the patch table much bigger. The current on based on Eddie's original is already 1.2 GB in size.